### ISSUE TITLE
Shortcuts and OSD Function not working 

### ISSUE THREAD

**Issue by ujjwalkirorimal** [seerge/g-helper#431]
**Describe the bug**
Shortcuts and OSD Function not working 

**Clear scenario to Reproduce**
1. Turn off ASUS Optimization service
2. Reboot my computer
3. Open G-Helper (Run on Startup)
4. Function button shortcuts and OSD doesn't work

**Expected behavior**
Function button shortcuts and OSD should work

**App Logs**
[log.txt](https://github.com/seerge/g-helper/files/11531452/log.txt)

**Screenshots**
![image](https://github.com/seerge/g-helper/assets/38243573/cf9ba62a-f6d3-427f-a0a0-2dae34993167)

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Laptop model - Asus TUF FX505-DT

**Asus software**
- Armoury crate uninstalled
- MyASUS uninstalled
- Other Asus services running in background - None

**Additional context**
Tried the apps provided in issue #429 but no success

**Comment by seerge** [comment]
@ujjwalkirorimal looks like you have trimmed log, as i don't see any info there about connected usb devices (keyboard) ? :)

You may try this build that will try to get all possible device IDs

[GHelper.zip](https://github.com/seerge/g-helper/files/11531570/GHelper.zip)

**Comment by ujjwalkirorimal** [comment]
Haven't trimmed the log though. I just deleted the old one and provided the fresh one.
Here is the log with above build. Since logs were same as before I tried cycling RGB brightness and color manually using app UI to get more info.
[log.txt](https://github.com/seerge/g-helper/files/11531682/log.txt)

**Comment by seerge** [comment]
@ujjwalkirorimal ok. 

It looks like on your model, hotkeys are handled in some other way (not known to me). As app simply don't see any USB device to listen to. 

For now you can keep using asus optimization service, as it should work as before.

**Comment by ujjwalkirorimal** [comment]
> 

Actually though when 0.66 was in pre-release phase. I tried the app once and osd and shortcuts used to work except touchpad one. Since it was in pre-rel phase I though touchpad thing would be fixed in rel but I guess everything totally borked?

**Comment by ujjwalkirorimal** [comment]
Here you go
[log.txt](https://github.com/seerge/g-helper/files/11531818/log.txt)

**Comment by seerge** [comment]
@ujjwalkirorimal well, then log from that "magical" build would be handy :) as it's pretty much clear now, that right now there is no usb device to listen to

**Comment by ujjwalkirorimal** [comment]
> @ujjwalkirorimal well, then log from that "magical" build would be handy :) as it's pretty much clear, that right now there is no usb device to listen to

That magical build worked because even though I stopped and disabled asus optimization, it was started once :p

**Comment by seerge** [comment]
@ujjwalkirorimal can you reproduce it again? By using 0.66 (or 0.67) ? :)

**Comment by Arakhor** [comment]
Same issue here, on G15 GA503RW
[log.txt](https://github.com/seerge/g-helper/files/11532054/log.txt)

**Comment by ujjwalkirorimal** [comment]
> @ujjwalkirorimal the only thing that comes to my mind, it could be that keyboards needs to be "woken up" from sleep before listening.
> 
> Try this build that will send init command to all usb devices [GHelper.zip](https://github.com/seerge/g-helper/files/11531983/GHelper.zip)

Didn't worked so as i said above, I started asus optimization service manually then stopped and disabled it. Then restarted the app.
[log.txt](https://github.com/seerge/g-helper/files/11532063/log.txt)

**Comment by seerge** [comment]
@ujjwalkirorimal then, since there is no reproduction scenario. And everything shows that in your case hotkeys don't come from USB device, i can't help much . You can keep using optimization service as before ofc :)

**Comment by Arakhor** [comment]
Sorry, the backlight control isn't working, the OSD shows up but it doesn't change the backlight slider inside GHelper, and even if it did, the slider has no effect either. If I enable Asus Optimization the backlight control works, slider still doesn't do anything. Weird because the slider at least used to work.

**Comment by seerge** [comment]
@Arakhor ok, so it's only backlight brightness that doesn't work ?

This is quite strange as app clearly sends payload to keyboard (and it's same device as being  used for animation / colors) :) 

```
22/05/2023 17:09:03: Key:197
22/05/2023 17:09:03: USB-KB 63:5D-BA-C5-C4-00
22/05/2023 17:09:03: Key:196
22/05/2023 17:09:03: USB-KB 63:5D-BA-C5-C4-01
```

Can you try this build with one small change : 
[GHelper.zip](https://github.com/seerge/g-helper/files/11533013/GHelper.zip)

**Comment by seerge** [comment]
@Arakhor slider won't change in realtime, i just didn't cover that :) (but it will if you reopen extra window)

As for backlight - are other open source apps like openRGB able to control brightness in your case?

**Comment by Arakhor** [comment]
I tried OpenRGB, same issue as with GHelper actually, color switching works, backlight brightness doesn't.
I can "sort of" lower the backlight by lowering value of a color in a HSV picker, possibly that's what Asus is doing on my model?

**Comment by seerge** [comment]
@Arakhor I don't know. It could be that on your specific model brightness is controlled in some not known way (so open rgb also can't access it). Therefore it's not much i can do here neither.

But do brightness adjustments work if asus optimization service is running?

**Comment by Arakhor** [comment]
Yes, with Asus Optimization everything works except for the slider.

**Comment by Arakhor** [comment]
I tried SignalRGB https://signalrgb.com/ and brightness control actually works in that app, device shows as ROG Strix Scar (2022).

**Comment by seerge** [comment]
@Arakhor ye, but signal rgb is not open source :) so i have no idea what it does differently. 
Can you try this build ?

[GHelper.zip](https://github.com/seerge/g-helper/files/11533760/GHelper.zip)

**Comment by Arakhor** [comment]
No difference sadly.
I've discovered that enabling Asus Optimization actually allows the slider to work though.
[log.txt](https://github.com/seerge/g-helper/files/11535852/log.txt)

**Comment by Arakhor** [comment]
Sorry for wasting your time, I've found out what the issue was. When I first disabled Asus Optimization I went into task scheduler and disabled "ASUS Optimization 36D18D69AFC3" as well, which opens ""C:\Windows\System32\DriverStore\FileRepository\asussci2.inf_amd64_e66da585a80ae7c5\ASUSOptimization\AsusHotkey.exe"". Turns out this exe is actually needed for the hotkeys to work. Enabling the service alone would let them work as well, so I didn't think of it before.

**Comment by seerge** [comment]
@Arakhor ok, so everything works with optimization service stopped?

**Comment by Arakhor** [comment]
Yes, everything works with service disabled and AsusHotkey.exe running

**Comment by seerge** [comment]
@Arakhor  ok , good.

@ujjwalkirorimal  any chance running that thing would help you as well?

In other case, it could be that on old TUF model that @ujjwalkirorimal has, keys work differently, and without physical access to device it's practically impossible to discover how. So for now we will have to depend on Optimization service there :)

Closing this issue then

**Comment by seerge** [comment]
@Arakhor just FYI in https://github.com/seerge/g-helper/releases/tag/v0.68 i have added some changes, that can make backlight controls working even w/o that AsusHotkey thing

**Comment by ujjwalkirorimal** [comment]
> @Arakhor just FYI in https://github.com/seerge/g-helper/releases/tag/v0.68 i have added some changes, that can make backlight controls working even w/o that AsusHotkey thing

Hey, Can you do something so that I can get g-helper OSD's even when asus-optimization is running? As I need it running for my hotkeys to work, OSD doesn't work as ghelper detects it.

**Comment by seerge** [comment]
@ujjwalkirorimal that's on purpose, as optimization service has own OSD, and showing double doesn't make much sense ?

**Comment by ujjwalkirorimal** [comment]
> @ujjwalkirorimal that's on purpose, as optimization service has own OSD, and showing double doesn't make much sense ?

Asus Optimization OSD never works for my laptop unless I have Myasus installed, thats why if OSD could have worked then it would be profit for me

**Comment by seerge** [comment]
@ujjwalkirorimal but then OSD would be just backlit notifications + touchpad ?

I have added one more attempt to detect your input device (to make hotkeys work) , can you check this build 
[GHelper.zip](https://github.com/seerge/g-helper/files/11545223/GHelper.zip)

**Comment by ujjwalkirorimal** [comment]
> @ujjwalkirorimal but then OSD would be just backlit notifications + touchpad ?
> 
> I have added one more attempt to detect your input device (to make hotkeys work) , can you check this build
> 
> [GHelper.zip](https://github.com/seerge/g-helper/files/11545283/GHelper.zip)
> 
> (and post a log here)
> 
> By the way, does linux work ok on your laptop? Especially backlight hotkeys
It would be fine with me as toggling touchpad without OSD feels odd to me :|

Didn't worked

[log.txt](https://github.com/seerge/g-helper/files/11545447/log.txt)

Its been a long while I have i installed linux on my laptop but if iirc it used to work fine

**Comment by ujjwalkirorimal** [comment]
> [GHelper.zip](https://github.com/seerge/g-helper/files/11546194/GHelper.zip)
> 
> Can you try this build (and post a log). I will try to listen for events literally from all ASUS usb devices in the system ...

Here you go

[log.txt](https://github.com/seerge/g-helper/files/11549788/log.txt)

**Comment by seerge** [comment]
@ujjwalkirorimal well, given that your laptop doesn't have any USB device that would emit / listen to events, no surprise. I assume touchpad is toggled in some other unknown to me way

Try this build if makes any difference 

[GHelper.zip](https://github.com/seerge/g-helper/files/11554600/GHelper.zip)

**Comment by ujjwalkirorimal** [comment]
> @ujjwalkirorimal well, given that your laptop doesn't have any USB device that would emit / listen to events, no surprise. I assume touchpad is toggled in some other unknown to me way
> 
> Try this build if makes any difference
> 
> [GHelper.zip](https://github.com/seerge/g-helper/files/11554600/GHelper.zip)

Nope, didn't worked

[log.txt](https://github.com/seerge/g-helper/files/11554627/log.txt)

**Comment by seerge** [comment]
@Arakhor and this 
[GHelper.zip](https://github.com/seerge/g-helper/files/11554641/GHelper.zip)

**Comment by seerge** [comment]
@ujjwalkirorimal and this
[GHelper.zip](https://github.com/seerge/g-helper/files/11554641/GHelper.zip)

**Comment by ujjwalkirorimal** [comment]
> @ujjwalkirorimal and this [GHelper.zip](https://github.com/seerge/g-helper/files/11554641/GHelper.zip)

Nope
[log.txt](https://github.com/seerge/g-helper/files/11554696/log.txt)

**Comment by seerge** [comment]
@ujjwalkirorimal well, then  you will have to live w/o touchpad disabling via hotkey :) Or keep using optimizatin service.

**Comment by ujjwalkirorimal** [comment]
> @ujjwalkirorimal well, then you will have to live w/o touchpad disabling via hotkey :) Or keep using optimizatin service.

Then I again request you to make osd work with asus optimization running xD

**Comment by seerge** [comment]
@ujjwalkirorimal can you try to run following in powershell as admin

``(Get-WmiObject -Namespace root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x00100011,1)``

(1 - to enable, 0 to disable)

It could be WMI touchpad toggle

**Comment by seerge** [comment]
@ujjwalkirorimal but does it disable touchpad if you will put 0?

**Comment by ujjwalkirorimal** [comment]
> @ujjwalkirorimal but does it disable touchpad if you will put 0?

Nope
